### PR TITLE
Add command line interface article

### LINKED ISSUE #28: Basic tutorial on Bonsai's Command Line Interface (CLI)

### PR DESCRIPTION
**Pull request by bruno-f-cruz**
Closes #28.

### CODE REVIEW

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:3]: Just a small suggestion for clarifying the writing: 

"The Bonsai CLI (Command-Line Interface) makes it possible to launch workflows from the OS command-line tool. In addition to opening workflows with the default graphic interface, the CLI enables the use of other advanced features, such as "no-editor" mode, initialization of variable's values, and specifying layouts."

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:9]: Another small suggestion for greater clarity: 

"The Bonsai CLI can be called by targeting the ```Bonsai.exe``` executable. To make sure you are able to call ```Bonsai``` from the command line, add it to your path using the following command:"

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:14]: Do you mean that you can simply call Bonsai from command line after adding ```pathToBonsaiExe``` to the System Path variable?

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:20]: I assume this will only work after Bonsai has been added to the System Path variable, either by adding it using the command ```pathToBonsaiExe\Bonsai.exe``` or by adding it through the System Path settings menu?

Also, I assume this only works on Windows operating systems?

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:26]: Edit for clarity: 

"To open an existing workflow, such as ```myWorkflow.bonsai```, use the syntax:"

or 

"To open an existing workflow (i.e. ```myWorkflow.bonsai```), use the syntax:"

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:42]: Edit for clarity: 

"Adding flags will change the output behavior of Bonsai CLI. For instance, to launch and immediately start a workflow from the command line, we can add the ```--start``` flag:"

File: articles/cli.md
Comment (bruno-f-cruz) [reply articles/cli.md:14]: Yes, if the directory ```pathToBonsaiExe``` exists in windows path, it will find Bonsai.exe and call it when you simply type "Bonsai" in the command line.

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:36]: is this an example that you are adding to the repo, so that people can actually run it? Otherwise I'm not sure what is the added benefit of having this additional example...

File: articles/cli.md
Comment (bruno-f-cruz) [reply articles/cli.md:20]: The command ```pathToBonsaiExe\Bonsai.exe``` is simply a way to call bonsai without adding it to the system variables. It works by calling Bonsai.exe using its full path. Alternatively, you can add ```pathToBonsaiExe``` to the Path as a mentioned previously.

I think that, for now, we should probably stick with windows only. Some of this syntax is likely to change in Linux anyway.

File: articles/cli.md
Comment (bruno-f-cruz) [reply articles/cli.md:36]: Sure, I was debating whether to add it or not since there is no file structure for workflows yet. What was the solution we came up with again? Simply opening a ""workflows" folder?

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:58]: Hmm, we should decide whether we refer to users as "the user" or simply directly, with "you"...this would go into the style guide of course. 

@glopesdev, any opinions?

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:64]: Edit for clarity: 

"Upon the successful launch of your workflow, a Bonsai icon will appear in your system tray, which can be used to stop the workflow or open other visualizers."

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:68]: Edit for clarity: 

"It is worth noting that only assigned visualizers will be available via this list."

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:65]: Edit for clarity: 

"Right click the node of interest;"

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:66]: Edit for clarity: 

"Under menu item "Show Visualizer", select any visualizer other than "None" "

File: articles/cli.md
Comment (bruno-f-cruz) [reply articles/cli.md]: I ended up making the article agnostic to a specific workflow.

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:75]: I'm not entirely sure what you are trying to say here...here's a possible edit for clarity (but please let me know if I'm misunderstanding what you are trying to say!): 

"One of the most useful features of the CLI is the ability to pass different values to different nodes before running a workflow. However, only "externalized properties" can be set in this way from the command line."

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:79]: Edit for clarity:

"Right click the node of interest;"

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:80]: Edit for clarity: 

"Under menu item "Externalize Property", select the property of interest "

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:84]: Edit for clarity: 

"A new node will appear with a dashed line connecting it to the original node of interest."

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:85]: What is meant by "in the same level"?

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:118]: Edit for clarity: 

"As of Bonsai version 2.6.1, it is possible to specify the layout file with which to run the workflow, by using the ```--no-editor mode```. This is especially useful when the same workflow is meant to be run several times in parallel (e.g. multiple identical behavior rigs)."

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:129]: Edit for clarity: 

"Currently, the easiest way to generate multiple layouts for a given script is to do the following:"

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:131]: Edit for clarity: 

"Place the windows in the desired layout;"

File: articles/cli.md
Comment (bruno-f-cruz) [reply articles/cli.md:85]: I was debating what word to use here. Basically, once you group nodes, you need to use the dot syntax, but it also allows you to have properties with the same name as long as they are not inside the same group. I dont want to use the word nested, since that is somewhat related with ```NestedWorkflow```. I could simply write:

"No two externalized properties, in the same group, are allowed to share the same name."

Would this be clearer?

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:134]: Edit for clarity: 

"This will create a ```*.bonsai.layout``` file with the same name as the workflow. This layout file should then be renamed to prevent it from being overwritten."

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:149]: Edit for clarity: 

"Additional flags are available, and can be used similarly to the ones introduced above. E.g.:"

File: articles/cli.md
Comment (taunsquared) [inline articles/cli.md:157]: Edit for brevity (it's pretty obvious that each flag is listed with a short description): 

"Currently available flags:"

File: articles/cli.md
Comment (glopesdev) [reply articles/cli.md:51]: In these cases, I tend to prefer using "you" as it strikes a more personal tone for communication. 👍

File: articles/cli.md
Comment (glopesdev) [reply articles/cli.md]: An alternative would be to simply link directly to the article explaining how externalized properties work in detail, and just describe here how the CLI interacts with these properties.

Comment (bruno-f-cruz) [reply articles/cli.md]: Should I just leave it as is for now (check the latest commit) and perhaps update it later once such an article has been written?

**Comment by glopesdev** [comment]
Formatting recommendations:
 * replace `~/` with local relative path `../`
 * remove `>` from the command-line snippets

**Review by glopesdev — APPROVED** [review]
Rebased the PR and added all revisions directly in branch before squash and merge.

**Pull request merged**
